zebra-rpc/src/methods/get_block_template_rpcs.rs

```diff
@@ -1101,10 +1101,12 @@ where
 
     async fn get_peer_info(&self) -> Result<Vec<PeerInfo>> {
         let address_book = self.address_book.clone();
+
         Ok(address_book
             .recently_live_peers(chrono::Utc::now())
             .into_iter()
-            .map(PeerInfo::from)
+            .filter(|meta_addr| meta_addr.has_connection_recently_responded(chrono::Utc::now()))
```

Comment:
`recently_live_peers()` shouldn't return any `MetaAddr` items that haven't recently responded, it's even more constrained, it also checks the last connection state.

```suggestion
```

Reply:
This is a correct suggestion but the mock used in tests was not doing the same causing a test to fail.
Fixed in https://github.com/ZcashFoundation/zebra/pull/9214/commits/835cca42052aa05cf11c0a5f7bd3ec01a44a803a to follow the same behaviour.